lib/matplotlib/axes/_base.py

```diff
@@ -4141,22 +4149,37 @@ def get_tightbbox(self, renderer, call_axes_locator=True,
             self.apply_aspect()
 
         if self.axison:
-            bb_xaxis = self.xaxis.get_tightbbox(renderer)
+            igl = 'x' if for_layout_only else None
+            try:
+                bb_xaxis = self.xaxis.get_tightbbox(renderer, ignore_label=igl)
+            except TypeError as e:
+                # in case downstream library has redefined axis:
+                bb_xaxis = self.xaxis.get_tightbbox(renderer)
             if bb_xaxis:
                 bb.append(bb_xaxis)
 
-            bb_yaxis = self.yaxis.get_tightbbox(renderer)
+            igl = 'y' if for_layout_only else None
+            try:
+                bb_yaxis = self.yaxis.get_tightbbox(renderer, ignore_label=igl)
+            except TypeError as e:
+                # in case downstream library has redefined axis:
+                bb_xaxis = self.yaxis.get_tightbbox(renderer)
```

Comment:
Do we want to warn so down-stream knows they can make this signature change?

Reply:
I don't know that we should warn.  But now I'm wondering if passing kwords around makes sense....

Reply:
An alternative is we provide new methods for these.  i.e. `get_tightbbox_forlayout` that does the "right" thing.    A second alternative is that the layout engines back out the correct info themselves, but that seems hard.

Reply:
I think the way this is done now is probably still the best, but other solutions welcome.

Reply:
I guess you may need something similar to #12635?

Reply:
We didn't warn in #12635 so we should probably stay consistent about that.

Reply:
I think since there is nothing the end user can do about the warning, and downstream libraries don't *need* this feature not warning is right.  It may be hard to advertise for downstream libraries, but its not a super crucial feature in my opinion, just a nicety.